### PR TITLE
Ft40-a: Add Login Page, Authentication Service and Upgrade to Angular 10

### PR DESCRIPTION
**Pull request by noce2**
## Purpose

- Adds a login page to the application
- Adds an authentication service with temporary users
- Upgrades the application to Angular 10 and Clarity 4+

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

- Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

- Test the code

```
npm run test
npm run e2e-local
```

## What to Check

Verify that the following are valid

- `npm run test`
```
Executed 52 of 59 (skipped 7) SUCCESS (2.029 secs / 1.696 secs)
TOTAL: 52 SUCCESS
```

- `npm run e2e-local`
```
[1] Executed 12 of 12 specs SUCCESS in 11 secs.
[1] [07:14:16] I/launcher - 0 instance(s) of WebDriver still running
[1] [07:14:16] I/launcher - chrome #01 passed
```

### CODE REVIEW

**Review by aakano684 — DISMISSED** [review]
LGTM. Good work, looks like we're ready to plug in Auth0 authentication logic. Maybe in the future we should discuss version bumps to make sure we both agree on the reasoning  as those can have some sneaky cascading effects on unrelated components and we can do some more comprehensive manual checks to avoid those.

**Comment by noce2** [comment]
@aakano684 the e2e tests were failing so I needed to fix those. There was a breaking change in the installation location of chrome on the Azure DevOps container. Pointing this out because we will need to apply a similar fix, https://github.com/math-dojo/single-page-web-app/pull/63/commits/f76f80424ed9175197b3fa39b254c1699a4abae4 , to the Admin-UI project when next we open a PR.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -44,14 +44,20 @@ steps:
...
 
 - script: |
-    ./node_modules/.bin/webdriver-manager update --versions.chrome=83.0.4103.39
+
+    if not exist "C:\Program Files (x86)\Google" mkdir "C:\Program Files (x86)\Google"
+    if not exist "C:\Program Files (x86)\Google\Chrome" mkdir "C:\Program Files (x86)\Google\Chrome"
+    mklink /D "C:\Program Files (x86)\Google\Chrome\Application" "C:\Program Files\Google\Chrome\Application"
+
+    ./node_modules/.bin/webdriver-manager update --versions.chrome=85.0.4183.38
   displayName: 'add chrome driver to node directory'
```
Comment (aakano684) [inline azure-pipelines.yml:53]: can you briefly explain what the bug was and how these lines fix it, for my own benefit mostly

**Review by aakano684 — COMMENTED** [review]
Happy to approve once I understand the fix you applied to the yaml. The rest seems fine

**Comment by noce2** [comment]
> Happy to approve once I understand the fix you applied to the yaml. The rest seems fine

Sure, the fix was f76f80424ed9175197b3fa39b254c1699a4abae4. In this, I added a symbolic link to where the new location for the Google Chrome binary.

**Review by aakano684 — APPROVED** [review]
LGTM

**Pull request merged**